src/bigquery.ts

```diff
@@ -729,6 +732,60 @@ export class BigQuery extends common.Service {
     return BigQuery.geography(value);
   }
 
+  /**
+   * Return a value's provided type.
+   *
+   * @private
+   *
+   * @throws {error} If the type provided is invalid.
+   *
+   * @see [Data Type]{@link https://cloud.google.com/bigquery/data-types}
+   *
+   * @param {*} providedType The type.
+   * @returns {string} The valid type provided.
+   */
+  // tslint:disable-next-line no-any
+  static getProvidedType_(providedType: any): ValueType {
+    const VALID_TYPES = [
+      'DATE',
+      'DATETIME',
+      'TIME',
+      'TIMESTAMP',
```

Comment:
not a blocker, but I wonder if there's any way we could use the `IStandardSqlDataType` from `./src/types.d.ts` for this list?

We might also consider this as the type for `providedType` rather than `any`.

Reply:
Could you possibly give me an example of what that would look like? I still have some issues with with types in TypeScript

Reply:
Also, I'm curious of you could take a peek at the tests. I feel like this isn't actually working like I want it to

Reply:
Digging into it, there might not be an easy way to do this, what I might just do is add a comment that says the list of types can be found in `/src/types.d.ts`, my only other question, do we need to allow the other enums listed:

```
      | 'NUMERIC'
      | 'ARRAY'
      | 'STRUCT';
```?

Reply:
Okay, got it.